File: lms-frontend/src/employee/services/workModeService.js
// workModeService.js

import api from "../../api/axios";

const BASE_URL = "/api/work-modes";

const getAllWorkModes = () => {

    return api.get(
        BASE_URL
    );
};

const getWorkModeById = (modeId) => {

    return api.get(
        `${BASE_URL}/${modeId}`
    );
};

const createWorkMode = (workModeData) => {

    return api.post(
        BASE_URL,
        workModeData
    );
};

const updateWorkMode = (
    modeId,
    workModeData
) => {

    return api.put(
        `${BASE_URL}/${modeId}`,
        workModeData
    );
};

const deleteWorkMode = (modeId) => {

    return api.delete(
        `${BASE_URL}/${modeId}`
    );
};

const workModeService = {

    getAllWorkModes,
    getWorkModeById,
    createWorkMode,
    updateWorkMode,
    deleteWorkMode
};

export default workModeService;

File: backend/src/main/java/com/lms_erp/employee/service/WorkModeService.java
package com.lms_erp.employee.service;

import com.lms_erp.employee.dto.WorkModeRequest;
import com.lms_erp.employee.dto.WorkModeResponse;

import java.util.List;

public interface WorkModeService {

    WorkModeResponse createWorkMode(WorkModeRequest request);

    WorkModeResponse updateWorkMode(Long modeId,
                                    WorkModeRequest request);

    WorkModeResponse getWorkModeById(Long modeId);

    List<WorkModeResponse> getAllWorkModes();

    void deleteWorkMode(Long modeId);
}